src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1573,12 +1598,15 @@ public bool DoIndexingWork(IndexingStatsScope stats, CancellationToken cancellat
 
                             tx.Commit();
                             SlowWriteNotification.Notify(commitStats, DocumentDatabase);
-
                             stats.RecordCommitStats(commitStats.NumberOfModifiedPages, commitStats.NumberOf4KbsWrittenToDisk);
                         }
                     }
                     catch
                     {
+                        // need to call it here to the let the index continue running
+                        // we'll stop when we reach the index error threshold 
+                        _mre.Set();
```

Comment:
This is called on _any_ exception, but isn't the intent just for disk full?

Reply:
This is for any exception. We have error handling and after a couple we set the index as errored. Unless we do that the index just stops

Reply:
This is okay for me but I'm a bit worried that we'll error indexes very quickly after getting any error here  (e.g. what is the chance we'll recover from the disk full exception, we'll repeat it 10 times and mark the index as errored).

I wonder if we should have a notion of fallback mode - e.g. after getting end of disk space then wait 10 seconds, and run indexing then.. next wait 20 seconds etc until the max fallback mode duration.

Reply:
Sounds good to me.
We should also force a cleanup of our state in this case, so a flush / sync of Voron and then ask to remove all the old recylced journals, old scratches, etc.

Reply:
done